core/math/a_star.cpp

```diff
@@ -193,10 +218,15 @@ PoolVector<int> AStar::get_point_connections(int p_id) {
 	return point_list;
 }
 
-bool AStar::are_points_connected(int p_id, int p_with_id) const {
+bool AStar::are_points_connected(int p_id, int p_with_id, bool bidirectional) const {
 
 	Segment s(p_id, p_with_id);
-	return segments.has(s);
+	const Set<Segment>::Element *element = segments.find(s);
+	int direction = bidirectional ?
+							0 : // Use 0 to accept any existing segment
+							(s.u == p_id ? Segment::FORWARD : Segment::BACKWARD);
```

Comment:
Can't `s.u == p_id ? Segment::FORWARD : Segment::BACKWARD` be replaced by `s.direction`?